pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/KitConfigurationDao.java

```diff
@@ -96,59 +114,65 @@ default long addCountryRule(long configId, String countryCode) {
+            kitConfigurations.add(getKitConfigurationForDto(kitConfigurationDto));
+        }
+        return kitConfigurations;
+    }
+
     default KitConfiguration getKitConfigurationForDto(KitConfigurationDto kitConfigurationDto) {
         Handle handle = getHandle();
         KitTypeDao kitTypeDao = handle.attach(KitTypeDao.class);
-        JdbiUmbrellaStudy jdbiUmbrellaStudy = handle.attach(JdbiUmbrellaStudy.class);
 
-        String kitTypeName = kitConfigurationDto.getKitType();
-        KitType kitType = kitTypeDao.getKitTypeByName(kitTypeName).get();
-        Collection<KitRule> kitRules = getTypedRulesForConfigurationId(handle, kitConfigurationDto.getId());
+        KitType kitType = kitTypeDao.getKitTypeById(kitConfigurationDto.getKitTypeId()).get();
+        Collection<KitRule> kitRules = findRulesByConfigId(kitConfigurationDto.getId());
         int numKits = (int) kitConfigurationDto.getNumberOfKits();
-        String guid = jdbiUmbrellaStudy.findGuidByStudyId(kitConfigurationDto.getStudyId());
+        String guid = kitConfigurationDto.getStudyGuid();
 
         return new KitConfiguration(kitConfigurationDto.getId(), numKits, kitType, guid, kitRules);
     }
 
-    default Collection<KitRule> getTypedRulesForConfigurationId(Handle handle, Long configurationId) {
-        Collection<KitRule> kitRules = new ArrayList<>();
-
-        JdbiKitRules jdbiKitRules = handle.attach(JdbiKitRules.class);
-        JdbiExpression jdbiExpression = handle.attach(JdbiExpression.class);
-        JdbiCountry jdbiCountry = handle.attach(JdbiCountry.class);
-
-        Collection<KitRuleDto> kitRuleDtos = getUntypedRulesForConfigurationId(configurationId, jdbiKitRules);
-
-        for (KitRuleDto kitRuleDto : kitRuleDtos) {
-            switch (kitRuleDto.getKitRuleType()) {
+    @SqlQuery("select kckr.kit_configuration_id,"
+            + "       kr.kit_rule_id,"
+            + "       krt.kit_rule_type_code as kit_rule_type,"
+            + "       pe.expression_text as pex_expression,"
+            + "       (select country_code from country where country_id = kc.country_id) as country_code,"
+            + "       (select group_concat(zip_code separator ',') from kit_zip_code where kit_rule_id = kr.kit_rule_id) as zip_codes"
+            + "  from kit_configuration__kit_rule as kckr"
+            + "  join kit_rule as kr on kr.kit_rule_id = kckr.kit_rule_id"
+            + "  join kit_rule_type as krt on krt.kit_rule_type_id = kr.kit_rule_type_id"
+            + "  left join kit_pex_rule as kp on kp.kit_rule_id = kr.kit_rule_id"
+            + "  left join expression as pe on pe.expression_id = kp.expression_id"
+            + "  left join kit_country_rule as kc on kc.kit_rule_id = kr.kit_rule_id"
+            + " where kckr.kit_configuration_id = :configId")
+    @UseRowReducer(KitRuleReducer.class)
+    List<KitRule> findRulesByConfigId(@Bind("configId") long configId);
+
```

Comment:
We're going to be querying these more often now, so did some minimal refactors to optimize the queries. 
There's more we can do later.